plugins/techdocs-node/src/stages/publish/awsS3.ts

```diff
@@ -183,20 +191,30 @@ export class AwsS3Publish implements PublisherBase {
     };
   }
 
-  private static buildCredentials(
+  private static async buildCredentials(
+    credsProvider: AwsCredentialsProvider,
+    accountId?: string,
     config?: Config,
     region?: string,
-  ): CredentialProvider {
+  ): Promise<AwsCredentialIdentityProvider> {
+    // Pull credentials for the specified account ID from the 'aws' config section
+    if (accountId) {
+      return (await credsProvider.getCredentials({ accountId })).provider;
+    }
+
+    // Fall back to the default credential chain if neither account ID
+    // nor explicit credentials are provided
     if (!config) {
-      return fromNodeProviderChain();
+      return (await credsProvider.getCredentials()).provider;
     }
 
+    // Pull credentials from the techdocs config section (deprecated)
```

Comment:
Should we perhaps invert this so that the deprecated method is check and used first, and then move on to the new provider?

I do feel we should be able to express this as a single call to `.getCredentials(...)` with all the information that's available. It gets a bit tedious to have to wrap it up in more logic. It does look like that's the case though, just a `return ...getCredentials({ accountId })...` at the end

Reply:
This logic would be a lot less complicated if this was a brand-new plugin without legacy config options to deal with, so I think it does look worse than it is in the Techdocs implementation.  In a new plugin, this entire method would be replaced with:
```ts
      const creds = (await credsProvider.getCredentials({
          accountId: config.getOptionalString('techdocs.publisher.awsS3.accountId'),
      })).provider;
```